internal/services/dispatcher/server.go

```diff
@@ -754,6 +755,25 @@ func (s *sendTimeFilter) canSend() bool {
 	return true
 }
 
+func calculateResultsSize(results []*contracts.StepRunResult) int64 {
+	var totalSize int64
+
+	for _, result := range results {
+		// Size of the pointer
+		totalSize += int64(unsafe.Sizeof(result))
```

Comment:
I'm not sure why we're using `unsafe` here -- this doesn't seem to correspond precisely to the size that we'd see when sent with grpc anyway, so why not just estimate using `len(*result.Output)` anyway?